tensorflow/core/platform/s3/aws_crypto.cc

```diff
@@ -100,6 +101,22 @@ class AWSSha256OpenSSLImpl : public Aws::Utils::Crypto::Hash {
   }
 };
 
+class AWSSecureRandomBytesImpl : public Aws::Utils::Crypto::SecureRandomBytes {
+public:
+  AWSSecureRandomBytesImpl() {}
+  virtual ~AWSSecureRandomBytesImpl() = default;
+  virtual void GetBytes(unsigned char* buffer, size_t bufferSize) override {
+    assert(buffer);
```

Comment:
I'm concerned about this `assert` here. We have some policy to not add `assert`s/`CHECK`s into code unless really needed. But let's keep this as it is now and see how the internal review goes.

Reply:
Okay